Fix flaky tests

### Environmental Information
ci test

- Onadata version: master

### Problem description
Fix these flaky tests
```
FAIL: test_api (onadata.apps.main.tests.test_form_api.TestFormAPI)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/onaio/onadata/onadata/apps/main/tests/test_form_api.py", line 44, in test_api
    self.assertEqual(find_d, data)
AssertionError: {'_id': 252, '_tags': [], '_uuid': '5b2cc31[617 chars]one'} != {'transport/available_transportation_types_[617 chars]alse}
Diff is 861 characters long. Set self.maxDiff to None to see it.
======================================================================
FAIL: test_retrieve_records_using_list_of_queries (onadata.apps.viewer.tests.test_parsed_instance.TestParsedInstance)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/onaio/onadata/onadata/apps/viewer/tests/test_parsed_instance.py", line 132, in test_retrieve_records_using_list_of_queries
    self.assertEqual(2, records.count())
AssertionError: 2 != 1
```

### Expected behavior
Test run consistently

### Steps to reproduce the behavior
ci test build

### Additional Information
None
_Logs, [related issues](github.com/onaio/onadata/issues), weird / out of place occurrences, local settings, possible approach to solving this..._

--- a/onadata/apps/main/tests/test_base.py
+++ b/onadata/apps/main/tests/test_base.py
@@ -35,7 +35,7 @@ from onadata.libs.utils.user_auth import get_user_default_project
 
 
 class TestBase(PyxformMarkdown, TransactionTestCase):
-
+    maxDiff = None
     surveys = ['transport_2011-07-25_19-05-49',
                'transport_2011-07-25_19-05-36',
                'transport_2011-07-25_19-06-01',

--- a/onadata/apps/main/tests/test_form_api.py
+++ b/onadata/apps/main/tests/test_form_api.py
@@ -34,8 +34,7 @@ class TestFormAPI(TestBase):
         request.user = self.user
         response = api(request, self.user.username, self.xform.id_string)
         self.assertEqual(response.status_code, 200)
-        data = dict_for_mongo_without_userform_id(
-            self.xform.instances.all()[0].parsed_instance)
+        data = self.xform.instances.all()[0].json
         find_d = json.loads(response.content)[0]
 
         # ensure all strings are unicode

--- a/onadata/apps/viewer/tests/test_parsed_instance.py
+++ b/onadata/apps/viewer/tests/test_parsed_instance.py
@@ -90,7 +90,7 @@ class TestParsedInstance(TestBase):
 
         instances = Instance.objects.filter(
             xform__id_string=self.xform.id_string
-        )
+        ).order_by('id')
         instances_count = instances.count()
         self.assertEqual(instances_count, 4)
 
@@ -132,10 +132,14 @@ class TestParsedInstance(TestBase):
         instances_debug = list(
             self.xform.instances.filter(user__username='bob').values_list(
                 'json', flat=True))
+        all_instances_debug = list(
+            instances.values_list('json', flat=True)
+        )
         self.assertEqual(
             2, records.count(),
             'Fields do not have ambulance. '
-            f'Fields submitted by bob are {instances_debug}')
+            f'Fields submitted by bob are {instances_debug}.'
+            f'All instances {all_instances_debug}')
 
     def test_parse_sort_fields_function(self):
         """

--- a/onadata/apps/viewer/tests/test_parsed_instance.py
+++ b/onadata/apps/viewer/tests/test_parsed_instance.py
@@ -129,7 +129,13 @@ class TestParsedInstance(TestBase):
         sql, params, records = get_sql_with_params(
             xform=self.xform, query=[{'_submitted_by': 'bob'}, 'ambulance']
         )
-        self.assertEqual(2, records.count())
+        instances_debug = list(
+            self.xform.instances.filter(user__username='bob').values_list(
+                'json', flat=True))
+        self.assertEqual(
+            2, records.count(),
+            'Fields do not have ambulance. '
+            f'Fields submitted by bob are {instances_debug}')
 
     def test_parse_sort_fields_function(self):
         """

